common/math/integer.go

```diff
@@ -0,0 +1,25 @@
+package math
+
+import gmath "math"
+
+/*
+ * NOTE: The following methods need to be optimised using either bit checking or asm
+ */
+
+// IsMinSafe returns whether the subtraction is safe and does not overflow.
+func SafeSub(x, y uint64) (uint64, bool) {
+	return x - y, x < y
+}
+
+// IsAddSafe returns whether the addition is safe and does not overflow.
+func SafeAdd(x, y uint64) (uint64, bool) {
+	return x + y, y > gmath.MaxUint64-x
+}
+
+// IsAddSafe returns whether the multiplication is safe and does not overflow.
```

Comment:
> // IsAddSafe returns whether the multiplication is safe and does not overflow.

Should be

// SafeMul returns multiplication result and whether overflow occurred.